worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlayerActor.java

```diff
@@ -520,22 +531,71 @@ public Region getLargestRegion() {
 
     @Override
     public void setSelection(Region region) {
-        RegionSelector selector;
-        if (region instanceof ConvexPolyhedralRegion) {
-            selector = new ConvexPolyhedralRegionSelector((ConvexPolyhedralRegion) region);
-        } else if (region instanceof CylinderRegion) {
-            selector = new CylinderRegionSelector((CylinderRegion) region);
-        } else if (region instanceof Polygonal2DRegion) {
-            selector = new Polygonal2DRegionSelector((Polygonal2DRegion) region);
-        } else {
-            selector = new CuboidRegionSelector(null, region.getMinimumPoint(),
-                    region.getMaximumPoint()
-            );
-        }
+        RegionSelector selector = switch (region) {
+            case ConvexPolyhedralRegion blockVector3s -> new ConvexPolyhedralRegionSelector(blockVector3s);
+            case CylinderRegion blockVector3s -> new CylinderRegionSelector(blockVector3s);
+            case Polygonal2DRegion blockVector3s -> new Polygonal2DRegionSelector(blockVector3s);
+            default -> new CuboidRegionSelector(null, region.getMinimumPoint(), region.getMaximumPoint());
+        };
         selector.setWorld(region.getWorld());
 
         getSession().setRegionSelector(getWorld(), selector);
     }
+
+    @Override
+    public void loadClipboardFromDisk() {
+        if (clipboardLoading.isWriteLocked()) {
```

Comment:
It looks like we only use the write lock. Can't we use a different lock then (or a `Semaphore` if you want to unlock from a different thread)? Besides that, I think there might be a race condition where the write lock is not locked at this call but below the if. If that happens on the main thread, we stall the main thread.